tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -246,14 +267,17 @@ private ExpressionEvaluationResult Evaluate(DebuggerSnapshotCreator snapshotCrea
                 !evaluationResult.HasError &&
                 !ProbeRateLimiter.Instance.Sample(ProbeInfo.ProbeId))
             {
-                snapshotCreator.Stop();
                 shouldStopCapture = true;
                 return evaluationResult;
             }
 
             if (evaluationResult.Metric.HasValue)
             {
-                LiveDebugger.Instance.SendMetrics();
+                LiveDebugger.Instance.SendMetrics(ProbeInfo.MetricKind.Value, ProbeInfo.MetricName, evaluationResult.Metric.Value);
+                // snapshot creator is created for all probes in the method invokers,
+                // if it is a metric probe, once we sent the value, we can stop the invokers working and dispose the snapshot creator
+                snapshotCreator.Dispose();
+                shouldStopCapture = true;
```

Comment:
It's not something to fix for now but maybe remember in the refactoring phase, it's a bit confusing that we are invalidating the state(s) (marking `IsActive` as false) since at the beginning it's also served the scenario when an exception got thrown from inside our instrumentation but now it serves a different purpose.

Reply:
Agree